Examples/Examples.playground/Pages/03-UIKitExtensions.xcplaygroundpage/Contents.swift

```diff
@@ -102,4 +102,31 @@ view.cornerRadius = 30
 // Add shadow to view
 view.addShadow(ofColor: .black, radius: 3, opacity: 0.5)
 
+//: UILabel extension
+
+let string = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
+
+let attributes = [NSAttributedString.Key.foregroundColor: UIColor.red, NSAttributedString.Key.font: UIFont.boldSystemFont(ofSize: 16) ]
+
+let attributedString = NSAttributedString(string: string, attributes: attributes)
+
+let label = UILabel()
+
+label.numberOfLines = 20
+
+label.attributedText = attributedString
+
+let singleLineSize = UILabel.size(thatFitsAttributedString: attributedString, withConstraints: CGSize(width: 100, height: CGFloat.greatestFiniteMagnitude))
```

Comment:
@daliadanila 
Throwing an idea here ... what do you think about an API like 
```swift
extension UILabel {
   func adjustToFit(string: String,  limitedBy size: CGSize) -> CGSize { ... }
}
```
Most use cases I think probably will use the size to adjust the label anyway ... 
What do you think?

Reply:
@LucianoPAlmeida That's how I thought about it initially, but both methods from CoreText use the NSAttributedString that have style info in it.
Unless I'd create a local instance of NSAttributedString that contains all extracted properties from UILabel (font, size, foregroundColor, etc) 🤔

Reply:
Humm, I thought it was just font, but even in that case we can get this information from the self label object at least font size and colors so we build our own attributed string internally... what else it is used for calculate this? 
And also we can have both API’s string and attract string :)

Reply:
There are around 30 different properties style related that the attributed string can have.  Found there's a handy property on the label called `attributedText` that does exactly that! 😄
I'll change the extension to match your suggested signature.

Reply:
Let’s wait a bit for others opinions before changing :) what you think?

Reply:
Vc @omaralbeik @guykogus

Reply:
> Let’s wait a bit for others opinions before changing :) what you think?

No problem 😃

Reply:
@LucianoPAlmeida I see your point, tbh I prefer a function that does not change the frame. but return the required size instead

Reply:
I'm just not sure I understand the purpose of this. View's have `sizeThatFits(_:)` and `sizeToFit()` functions. Why not just set the `attributedText` of a label and call one of those? Is your purpose to avoid creating a `UILabel`?

Reply:
@guykogus TBH I really don’t know, I personally never use something like this, but seem some people use something like this, so I figure it may be useful for some case :)

Reply:
@guykogus I'm not sure as well, but I remember having to deal with layouting `UILabels` a couple of years ago in one app where `sizeThatFits` and `sizeToFit` was very buggy in complex `UICollectionView` layouts

@daliadanila could you please explain where you use this extension? what benefits does it provide over `sizeThatFits` and `sizeToFit`?

Reply:
@tospery suggested this on the issue, maybe he can provide some input here too :)

Reply:
@guykogus @omaralbeik @LucianoPAlmeida  I'd think people might find it useful when they'd have to work with strings/labels that have complex styling. In the original issue that was requesting this feature, [TTTAttributedLabel](https://github.com/TTTAttributedLabel/TTTAttributedLabel/blob/master/TTTAttributedLabel/TTTAttributedLabel.m) was mentioned that is implementing a similar method 
（sizeThatFitsAttributedString:withConstraints:limitedToNumberOfLines. 

Maybe @tospery can help with a use case when you'd use the CT methods over sizeThatFits:/sizeToFit methods.

Reply:
`TTTAttributedLabel` was created before you could assign an attributed string to a label. If you're still using it for any purpose then for the love of all that's holy please stop! 😅

Reply:
>  I'd think people might find it useful when they'd have to work with strings/labels that have complex styling.

I think if `sizeToFit()` and `sizeThatFits` works in most cases and this is more used for very specific scenarios and complex things, I'm not sure we should add this here ... I mean we would increase the library with an addition that would be rarely be used if used... 
Let me know what you think @daliadanila @guykogus @omaralbeik :)

Reply:
Just making sure, everyone agrees or are we moving forward with this? :)